File: CafeEase/UI/cafeease_mobile/lib/models/review_insert_request.dart
class ReviewInsertRequest {
  final int productId;
  final int rating;
  final String comment;

  ReviewInsertRequest({
    required this.productId,
    required this.rating,
    required this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      "productId": productId,
      "rating": rating,
      "comment": comment,
    };
  }
}

File: CafeEase/CafeEase.Model/Requests/ReviewInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace CafeEase.Model.Requests
{
    public class ReviewInsertRequest
    {
        public int ProductId { get; set; }

        [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5.")]
        public int Rating { get; set; }
        public string? Comment { get; set; }
    }
}
